packages/astro/src/integrations/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { AddressInfo } from 'net';
+import type { ViteDevServer } from 'vite';
+import { AstroConfig, AstroRenderer } from '../@types/astro.js';
+import { mergeConfig } from '../core/config.js';
+
+export async function runHookConfigSetup({ config: _config, command }: { config: AstroConfig; command: 'dev' | 'build' }): Promise<AstroConfig> {
+	let updatedConfig: AstroConfig = { ..._config };
+	for (const integration of _config.integrations) {
+		if (integration.hooks['astro:config:setup']) {
+			await integration.hooks['astro:config:setup']({
+				config: updatedConfig,
+				command,
+				addRenderer(renderer: AstroRenderer) {
+					updatedConfig._ctx.renderers.push(renderer);
+				},
+				// TODO: Add support for `injectElement()` for full HTML element injection, not just scripts.
+				// This may require some refactoring of `scripts`, `styles`, and `links` into something
+				// more generalized. Consider the SSR use-case as well.
+				injectElement: () => {
+					throw new Error('TODO: Implement');
```

Comment:
Is the plan to merge without initial support for this hook? If so a better error message should be provided like:

```
injectElement is not currently supported.
```

Reply:
No need to ship a function that throws, I'll just remove it for now and leave the TODO in the type def. This isn't blocking, but it's definitely something that I'd love to get in soon (possibly with a better understanding of how it could work with SSR).